apps/client/src/pages/QuestionsPage.js

```diff
@@ -46,34 +50,78 @@ const QuestionsPage = ({ topic, checker }) => {
     });
 
   const onQuestionNext = (value) => {
-    if (question.options) {
+    // Provide the user answers to the `sttr-checker`
+    if (question.options && value) {
       question.setAnswer(value);
-    } else {
+    }
+    if (!question.options && value) {
       const responseObj = booleanOptions.find((o) => o.formValue === value);
       question.setAnswer(responseObj.value);
     }
 
-    if (needContactPermits()) {
+    // Store all answers in the session context
```

Comment:
Nice optimization would be to implement the syncronization logic inside the checker itself.
Could be an onSetAnser callback. Or give it a 'store' to put the data in and read from.

Reply:
Great!